core/src/test/java/com/sequenceiq/cloudbreak/service/LoadBalancerConfigConverterTest.java

```diff
@@ -181,6 +181,33 @@ public void testConvertAzureTargetGroup() {
         assertEquals(AVAILABILITY_SET_NAME, availabilitySets.get(0));
     }
 
+    @Test
+    public void testConvertAzureTargetGroupMissingAvailabilitySetInMetadata() {
+        Map<String, Object> azureParams = createAzureParams(1);
+        azureParams.put(AzureLoadBalancerMetadataView.getAvailabilitySetParam(PORT2), null);
+        CloudLoadBalancerMetadata cloudLoadBalancerMetadata = new CloudLoadBalancerMetadata.Builder()
+                .withParameters(azureParams)
+                .build();
+        TargetGroup targetGroup = new TargetGroup();
+        targetGroup.setType(TargetGroupType.KNOX);
+        TargetGroupPortPair portPair1 = new TargetGroupPortPair(PORT1, PORT2);
+        TargetGroupPortPair portPair2 = new TargetGroupPortPair(PORT2, PORT2);
+        TargetGroupPortPair portPair3 = new TargetGroupPortPair(PORT3, PORT3);
+
+        when(loadBalancerConfigService.getTargetGroupPortPairs(eq(targetGroup))).thenReturn(Set.of(portPair1, portPair2, portPair3));
+
+        TargetGroupConfigDbWrapper targetGroupConfigDbWrapper = underTest.convertTargetGroup(AZURE, cloudLoadBalancerMetadata, targetGroup);
+        assertNotNull(targetGroupConfigDbWrapper.getAzureConfig());
+        AzureTargetGroupConfigDb azureTargetGroupConfigDb = targetGroupConfigDbWrapper.getAzureConfig();
+        assertEquals(Set.of(PORT1, PORT2, PORT3), azureTargetGroupConfigDb.getPortAvailabilitySetMapping().keySet());
+        List<String> availabilitySets = azureTargetGroupConfigDb.getPortAvailabilitySetMapping().get(PORT1);
+        assertEquals(List.of(AVAILABILITY_SET_NAME), availabilitySets);
+        availabilitySets = azureTargetGroupConfigDb.getPortAvailabilitySetMapping().get(PORT2);
```

Comment:
nit: this could be a bit cleaner without reassigning availability sets all the time.